recipes-nemomobile/sensorfw/sensorfw-hybris-hal-plugins_git.bb

```diff
@@ -0,0 +1,18 @@
+require sensorfw.inc
+
```

Comment:
Shouldn't this be marked for machine specific recompilation ?

Reply:
We can. But I'm not entirely sure if we need it.
My thought process is as follows: All watches that don't need a patched `sensorfw` could use the generic `armv7vehf-neon` while the watches that need a patched version can use the one specific to a machine.
This would mean that https://github.com/AsteroidOS/meta-smartwatch/pull/81 needs the `PACKAGE_ARCH = "${MACHINE_ARCH}"` applied for some watches.

Or would it be better to make every version specific to a watch? Just apply `PACKAGE_ARCH = "${MACHINE_ARCH}"` here. Like your comment.

Reply:
Added anyway since it solves another issue where `libhybrissensorfw*` is installed in the deploy directory by the hal and the binder version. It resulted in the following issue:
```
ERROR: sensorfw-hybris-hal-plugins-+gitAUTOINC+2e53901599-r1 do_package_write_ipk: The recipe sensorfw-hybris-hal-plugins is trying to install files into a shared area when those files already exist. Those files and their manifest location are:
  /home/darrel/AsteroidOS/asteroid-20220512/build/tmp-glibc/deploy/ipk/sturgeon/libhybrissensorfw-qt5-dev_+git0+2e53901599-r1_sturgeon.ipk
    (matched in manifest-sturgeon-sensorfw-hybris-binder-plugins.package_write_ipk)
  /home/darrel/AsteroidOS/asteroid-20220512/build/tmp-glibc/deploy/ipk/sturgeon/libhybrissensorfw-qt5-1_+git0+2e53901599-r1_sturgeon.ipk
    (matched in manifest-sturgeon-sensorfw-hybris-binder-plugins.package_write_ipk)
  /home/darrel/AsteroidOS/asteroid-20220512/build/tmp-glibc/deploy/ipk/sturgeon/libhybrissensorfw-qt5-dbg_+git0+2e53901599-r1_sturgeon.ipk
    (matched in manifest-sturgeon-sensorfw-hybris-binder-plugins.package_write_ipk)
  /home/darrel/AsteroidOS/asteroid-20220512/build/tmp-glibc/deploy/ipk/sturgeon/libhybrissensorfw-qt5-src_+git0+2e53901599-r1_sturgeon.ipk
    (matched in manifest-sturgeon-sensorfw-hybris-binder-plugins.package_write_ipk)
Please verify which recipe should provide the above files.
```

There are two solutions to this:
- Use `SSTATE_DUPWHITELIST = "${DEPLOY_DIR}/"` to allow the dupplicate files to exist. This probably introduces an issue with the package manager though since we can't reliably say what the ipk targets (hal or binder?).
- Make the package machine specific. Only one of the variants should ever be installed for a specific target. This also removes the need for making the package machine specific when a platform overrides it.

Reply:
What about a third solution where you'd just rename the libhybrissensorfw* files to libhybrisbindersensorfw* in the binder recipe ? Then both packages should be able to be installed in parallel, no ? Maybe the sensorfw config would need a slight variation to load the binder file but this should be manageable

Reply:
So, it seems that your proposed solution will not work, since the simply renaming the .so file to something else. This results in the exact same error (100% identical, the .ipk has the exact same name still).

The reason this does not work, is because the package is renamed by Yocto from `sensorfw-hybris-hal-plugins` (and `sensorfw-hybris-binder-plugins`) to `libhybrissensorfw-qt5-1`.
The reason this happens has something to do with how Yocto generates package names. Depending on the contents of the package (https://github.com/openembedded/openembedded-core/blob/master/meta/classes/debian.bbclass#L112) it will use `objdump` if it's a library to get the `SONAME` from the .so. Which is as follows:
```
arm-linux-gnueabi-objdump -p ..../build/tmp-glibc/work/beluga-oe-linux-gnueabi/sensorfw-hybris-binder-plugins/+gitAUTOINC+2e53901599-r1/packages-split/sensorfw-hybris-binder-plugins/usr/lib/libhybrisbindersensorfw-qt5.so.1.0.0

....

Dynamic Section:
  NEEDED               libsensordatatypes-qt5.so.1
  NEEDED               libgbinder.so.1
  NEEDED               libglib-2.0.so.0
  NEEDED               libQt5Core.so.5
  NEEDED               libstdc++.so.6
  NEEDED               libgcc_s.so.1
  NEEDED               libc.so.6
  SONAME               libhybrissensorfw-qt5.so.1
  INIT                 0x00003ca0
  FINI                 0x0000d630
  INIT_ARRAY           0x0001fdcc
  INIT_ARRAYSZ         0x00000004
  FINI_ARRAY           0x0001fdd0
  FINI_ARRAYSZ         0x00000004
  GNU_HASH             0x00000138
....
```

Then it simply renames the package in the `log.do_package` stage: `NOTE: debian: renaming sensorfw-hybris-binder-plugins to libhybrissensorfw-qt5-1`.
Which also results in a package renaming in the `log.do_package_write_ipk`: `NOTE: package name mapping done: sensorfw-hybris-binder-plugins -> libhybrissensorfw-qt5-1`. (https://github.com/openembedded/openembedded-core/blob/master/meta/classes/package.bbclass#L610)

Reply:
Now solved by setting `DEBIANNAME` so that the package isn't renamed.